src/App.tsx

```diff
@@ -1,72 +1,70 @@
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropdownMenu } from './components/DropdownMenu';
+import { Person } from './types/Person';
+import debounce from 'lodash.debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [title, setTitle] = useState(selectedPerson?.name || '');
+  const [appliedTitle, setAppliedTitle] = useState('');
+
+  const applyTitle = useCallback(debounce(setAppliedTitle, 300), []);
+
+  useEffect(() => {
+    setTitle(selectedPerson?.name || '');
+  }, [selectedPerson]);
+
+  const filteredUsers = useMemo(() => {
+    return [...peopleFromServer].filter(person =>
+      person.name.includes(appliedTitle),
+    );
+  }, [appliedTitle]);
+
+  const [inputFieldFocus, setInputFieldFocus] = useState(false);
+
+  function handleTitle(event: React.ChangeEvent<HTMLInputElement>) {
+    setTitle(event.target.value);
+    applyTitle(event.target.value);
```

Comment:
The `applyTitle` function is being called with `event.target.value`, but the debounce function should be used to handle the input changes. Ensure that spaces-only input is filtered out before applying the title.